modules/samples/watertank/component/plugin/src/main/java/org/homeautomation/watertank/plugin/impl/DeviceTypeManager.java

```diff
@@ -0,0 +1,257 @@
+            DeviceTypeDAO.beginTransaction();
+            status = deviceTypeDAO.getDeviceTypeDAO().updateDevice(device);
+            DeviceTypeDAO.commitTransaction();
+        } catch (DeviceMgtPluginException e) {
+            try {
+                DeviceTypeDAO.rollbackTransaction();
+            } catch (DeviceMgtPluginException iotDAOEx) {
+                String msg = "Error occurred while roll back the update device transaction :" + device.toString();
+                log.warn(msg, iotDAOEx);
+            }
+            String msg = "Error while updating the enrollment of the watertank device : " +
+                    device.getDeviceIdentifier();
+            log.error(msg, e);
+            throw new DeviceManagementException(msg, e);
+        }
+        return status;
+    }
+
+    @Override
+    public boolean disenrollDevice(DeviceIdentifier deviceId) throws DeviceManagementException {
+        boolean status;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Dis-enrolling watertank device : " + deviceId);
+            }
+            DeviceTypeDAO.beginTransaction();
+            status = deviceTypeDAO.getDeviceTypeDAO().deleteDevice(deviceId.getId());
+            DeviceTypeDAO.commitTransaction();
+        } catch (DeviceMgtPluginException e) {
+            try {
+                DeviceTypeDAO.rollbackTransaction();
+            } catch (DeviceMgtPluginException iotDAOEx) {
+                String msg = "Error occurred while roll back the device dis enrol transaction :" + deviceId.toString();
+                log.warn(msg, iotDAOEx);
+            }
+            String msg = "Error while removing the watertank device : " + deviceId.getId();
+            log.error(msg, e);
+            throw new DeviceManagementException(msg, e);
+        }
+        return status;
+    }
+
+    @Override
+    public boolean isEnrolled(DeviceIdentifier deviceId) throws DeviceManagementException {
+        boolean isEnrolled = false;
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Checking the enrollment of watertank device : " + deviceId.getId());
+            }
+            Device iotDevice =
+                    deviceTypeDAO.getDeviceTypeDAO().getDevice(deviceId.getId());
+            if (iotDevice != null) {
+                isEnrolled = true;
+            }
+        } catch (DeviceMgtPluginException e) {
+            String msg = "Error while checking the enrollment status of watertank device : " +
+                    deviceId.getId();
+            log.error(msg, e);
+            throw new DeviceManagementException(msg, e);
+        }
+        return isEnrolled;
+    }
+
+    @Override
+    public boolean isActive(DeviceIdentifier deviceId) throws DeviceManagementException {
+        return true;
+    }
+
+    @Override
+    public boolean setActive(DeviceIdentifier deviceId, boolean status)
+            throws DeviceManagementException {
+        return true;
```

Comment:
Fixed in PR  https://github.com/wso2/product-iots/pull/177 by throwing NotImplementedException as they are not implemented in this class.